detect_secrets/transformers/yaml.py

```diff
@@ -218,6 +227,41 @@ def _compose_node_shim(
 
         return cast(yaml.nodes.Node, node)
 
+    def _parse_flow_mapping_key_shim(
```

Comment:
I trust you that this works, but could you write some comments here explaining what this code does?

Reply:
Updated. Please let me know if that makes sense.